presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/TransformUncorrelatedScalarToJoin.java

```diff
@@ -66,5 +72,22 @@ public PlanNode visitApply(ApplyNode node, RewriteContext<PlanNode> context)
             }
             return rewrittenNode;
         }
+
+        private boolean hasScalarSubquery(ApplyNode node)
+        {
+            return node.getSubquery() instanceof EnforceSingleRowNode;
+        }
+
+        private boolean hasScalarAggregationSubquery(ApplyNode node)
+        {
+            return findFirstPlanNodeMatching(
```

Comment:
This is hard to read due to arguments being lined up with the chained calls. Unfortunately, there's no way to indent them differently in IntelliJ. 

I'd introduce a temporary variable to make it clear what's going on:

```java
Optional<PlanNode> aggregation = findFirstPlanNodeMatching(
        node.getSubquery(),
        AggregationNode.class::isInstance,
        Predicates.<PlanNode>isInstance(ProjectNode.class).or(LimitNode.class::isInstance));

return aggregation
        .map(AggregationNode.class::cast)
        .map(AggregationNode::getGroupBy)
        .map(List::isEmpty)
        .orElse(false);
```

Alternatively, we could introduce a builder-style matcher and rewrite it like:

```java
return matcher(node.getSubquery())
    .where(AggregationNode.class::isInstance)
    .skip(instanceOfAny(ProjectNode.class, LimitNode.class)
    .find()
    .map(AggregationNode.class::cast)
    .map(AggregationNode.getGroupBy)
    .map(List::isEmpty)
    .orElse(false)
```